serpens/pubsub.py

```diff
@@ -26,3 +26,32 @@ def publish_message(
     future = publisher.publish(topic, data=message, ordering_key=ordering_key, **attributes)
 
     return future.result()
+
+
+def publish_message_batch(topic: str, messages: List[Dict], ordering_key: str = "") -> List[str]:
+    publisher = pubsub_v1.PublisherClient()
+    results = []
+    endpoint = None
+
+    if ":" in topic:
+        topic, endpoint = topic.split(":")
+
+    for message in messages:
+        if not isinstance(message["body"], str):
+            message["body"] = json.dumps(message["body"], cls=SchemaEncoder)
+
+        body = message["body"].encode("utf-8")
+
+        if "attributes" not in message:
+            message["attributes"] = {}
+
+        if endpoint is not None:
+            message["attributes"]["endpoint"] = endpoint
+
+        future = publisher.publish(
+            topic, data=body, ordering_key=ordering_key, **message["attributes"]
+        )
+
+        results.append(future.result())
```

Comment:
Eu tenho a impressão que para enviar em lote você não deveria aguardar a resolução da promise retornada na chamada do método de publicação.

Pelo que vi na documentação por padrão ele já criaria um buffer para armazenar as mensagens e enviar em lote caso seja chamado o método de publicação inúmeras vezes seguidas até atingir uma das 3 condições, tamanho máximo de dados a serem enviados, tempo em que as mensagens estão aguardando o envio ou número máximo de mensagens.

Acho bom fazer testes para entender o funcionamento correto para o envio em lote, meu chute é que desta forma ele vai enviar uma mensagem de cada vez o que tornaria o processo muito mais lento.